src/intptrcast.rs

```diff
@@ -72,12 +80,41 @@ impl<'mir, 'tcx> GlobalStateInner {
                     None
                 }
             }
-        };
-        // Pointers created from integers are untagged.
-        Pointer::new(
-            alloc_id.map(|alloc_id| Tag { alloc_id, sb: SbTag::Untagged }),
-            Size::from_bytes(addr),
-        )
+        }
+        .and_then(|alloc_id| {
```

Comment:
We shouldn't need `and_then`; we are inside an `Option`-returning function so we can just use `?` to short-circuit `None`.